src/api/spec/features/webui/projects_spec.rb

```diff
@@ -34,8 +39,40 @@
     expect(find('#project_title').text).to eq("#{user.home_project_name}:coolstuff")
   end
 
-  it_behaves_like 'user tab' do
-    let(:project_path) { project_show_path(project: user_tab_user.home_project_name) }
-    let(:project) { Project.find_by_name(user_tab_user.home_project_name) }
+  describe "locked projects" do
+    let!(:locked_project) { create(:locked_project, name: "locked_project") }
+    let!(:relationship) { create(:relationship, project: locked_project, user: user) }
+
+    scenario "unlock project" do
+      login user
+
+      visit project_show_path(project: locked_project.name)
+      expect(page).to have_text("is locked")
+
+      click_link("Unlock project")
+      fill_in "comment", with: "Freedom at last!"
+      click_button("Ok")
+      expect(page).to have_text("Successfully unlocked project")
+
+      visit project_show_path(project: locked_project.name)
```

Comment:
Basically you could do everything in a before filter except checking the result, can't you?